doc/paper/paper.tex

```diff
@@ -0,0 +1,1012 @@
+    the database designer (i.e UDFs), and perhaps a more frequent need to
+    correlate multiple signals in a single search. The need for timestamp-based
+    search controls and filtering on high-level dimensions remains as great as
+    before.
+
+    Second, the meaning of \q{listing log lines} or \q{filling a screen with
+    logs} becomes a lot less clear, and is an interesting question in user
+    interface design. Log lines can still be listed as JSON, with bytes-valued
+    fields rendered in base64 or similar – but more creative approaches can
+    likely do better by incorporating images, pointcloud visualizations, maps,
+    audio, and more.
+
+
+    Third, text rendering is clearly insufficient for the \q{view in context}
+    function, which now represents replaying \q{what the robot saw} at the
+    associated time. Instead of a pinned message in a larger resultset, \q{view
+    in context} is implemented using a complex visualization application such as
+    Foxglove.
+
+    Beyond these parallels, there are other points of divergence:
+
+    \textbf{Different organizational profile.} Server application logs are
+    generally browsed only by infrastructure teams and application
+    administrators. The read activity on those databases is sparse and high
+    latencies are tolerable. In a robotics development organization, a
+    multimodal log database is a business-critical, user-facing service and
+    may be depended on by most of the development team outside of
+    infrastructure. High latencies in playback and search are a direct
+    impediment to product development speed.
+
+    \textbf{Different scales of volume.} Depending on what kind of data the user
+    is logging, multimodal data volumes can be orders of magnitude larger
+    than server application log volumes for the same organization. Large
+    producers may produce hundreds of terabytes or petabytes of logs per day.
+    This makes it relatively more important for administrators to have a
+    clear understanding of data access patterns, and granular control over
+    data retention.
+
+    \textbf{Support for distributed compute.} Typical server log databases do not
+    expect you to run distributed compute jobs on your log data and are not
+    designed with that use-case in mind, whereas in multimodal log management
+    it is a core workload.
+
+    \textbf{Support for statistical aggregates.} Multimodal logs are logs, but
+    in a lot of cases they are also similar to metrics. For some kinds of logs
+    it is attractive to be able to plot statistical aggregates such as
+    quantiles or averages of field values over selected windows of time. The
+    APM portions of Datadog enable this kind of operation at low latency over
+    trillions or quadrillions of points. To drive a responsive interface these
+    aggregations must rely on precomputation.
+
+    Of course, it can be assumed that in some large organizations these concerns
```

Comment:
Not sure this paragraph is needed. For brevity, we can move on to the next paragraph that dives into the challenges with large-scale multimodal log data (particularly seen with startups)